### DISCUSSION TITLE
30%+ CPU load on default example - just moving the mouse

### DISCUSSION THREAD

**Discussion by c64zottel** [emilk/egui#2162] [Q&A] [answered]
Hello,

I just downloaded and start the template - once in web mode and once in standalone mode.
I noticed when just moving the mouse (like an idiot) forth and back the load spikes beyond 30%.
A bit more specific:
google chrome: 100%
firefox: 30-40%
standalone: 30-40%

When I enable --release, the standalone version is doing better < 10%,
but ff is still up and shows: 
![image](https://user-images.githubusercontent.com/5398700/196392937-e0667993-9100-42f9-8f93-72b2b442b1f8.png)
Despite my effort in adding --release: trunk serve --release (or building first like trunk build --release)
The tool tip of "Debug build" mentions: "assertions enabled" (standalone version doesn't)
How do I disable this?

Having:
[profile.release]
opt-level = 3 # fast and small wasm
debug-assertions = false

does not help.

**Comment by emilk** [comment, marked as answer]
I tried this on `eframe_template`:

```
trunk build --release
(cd dist && basic-http-server --addr 127.0.0.1:8765 .)
```

![Screen Shot 2022-10-19 at 20 26 41](https://user-images.githubusercontent.com/1148717/196774140-0b3f9cf9-1572-4eab-a39a-f48bcc3c234f.png)

no "Debug build" warning.

Perhaps you need to do a hard refresh in your browser?

**Reply by c64zottel** [reply]
That did the trick. thx
Weirdly, the bad performance persists on ff.

**Reply by coderedart** [reply]
could be the caching service worker loading some cached build from the past. 

best way is to do `ctrl + F5` to make sure the latest version is being loaded from filesystem rather than the cached version.

**Reply by c64zottel** [reply]
Doesn't change anything.
But what I noticed, is that moving the mouse at the demo site (https://www.egui.rs/#demo) triggers a redraw.
I assume that is just more expensive under ff opposed to run it native.
